dbms/src/IO/Checksum/tests/gtest_dm_checksum_buffer.cpp

```diff
@@ -372,5 +372,115 @@ TEST_STACKED_SEEKING(CRC32)
 TEST_STACKED_SEEKING(CRC64)
 TEST_STACKED_SEEKING(City128)
 TEST_STACKED_SEEKING(XXH3)
-} // namespace tests
-} // namespace DB
+
+template <ChecksumAlgo D>
+void runCompressedSeekableReaderBufferTest()
+{
+    // Create a temporary file for testing
+    const std::string temp_file_path = "/tmp/tiflash_compressed_seek_test.dat";
+    SCOPE_EXIT({
+        Poco::File file(temp_file_path);
+        if (file.exists())
+            file.remove();
+    });
+    // Test data - create multiple blocks with different patterns
+    std::vector<std::string> test_blocks;
+
+    test_blocks = {
+        std::string(1500, 'A') + "BLOCK0_END",
+        std::string(800, 'B') + "BLOCK1_END",
+        "", // Block 2 is empty
+        "", // Block 3 is empty
+    };
```

Comment:
When will we write "empty block" to the `CompressedWriteBuffer<ChecksumWriteBuffr>`? When there are large number of "NULL" values in the block?

Reply:
Yes.